lib/syncthing/syncthing.go

```diff
@@ -475,3 +482,37 @@ func (e *controller) Shutdown() {
 func (e *controller) ExitUpgrading() {
 	e.Stop(ExitUpgrade)
 }
+
+type supervisor interface{ Services() []suture.Service }
+
+func printServiceTree(w io.Writer, sup supervisor, level int) {
+	printService(w, sup, level)
+
+	svcs := sup.Services()
+	sort.Slice(svcs, func(a, b int) bool {
+		return fmt.Sprint(svcs[a]) < fmt.Sprint(svcs[b])
+	})
+
+	for _, svc := range svcs {
+		if sub, ok := svc.(supervisor); ok {
+			printServiceTree(w, sub, level+1)
+		} else {
+			printService(w, svc, level+1)
+		}
+	}
+}
+
+func printService(w io.Writer, svc interface{}, level int) {
+	type errorer interface{ Error() error }
+
+	t := "-"
+	if _, ok := svc.(supervisor); ok {
+		t = "+"
+	}
+	fmt.Fprintln(w, strings.Repeat("  ", level), t, svc)
+	if es, ok := svc.(errorer); ok {
```

Comment:
You can just do: `foo.(interface { Error() error })` to avoid the type def.

Reply:
Yeah but is that any clearer? I dunno 🤷‍♂

Reply:
Probably, than a typedef in the function body.

Reply:
I don't care either way - if I needed to come up with any rationale I'd probably go with Audrius variant as it only appears once and thus when reading you don't need to search for the definition of `errorer`. That's however super contrived, it's pretty clear what's going on anyway

One might even throw in a third option if one wanted to bikeshed hard: Define the interface in lib/util or somewhere:
```
 $ grep -r "func() error" lib/ | wc -l
28
```
I hope it's clear I am not actually suggesting doing that for this PR though :)